jwala-common/src/main/java/com/cerner/jwala/common/FileUtility.java

```diff
@@ -100,7 +99,7 @@ private void closeJarFile(final JarFile jarFile) {
     public Set<String> getZipRootDirs(final String zipFilename) {
         final Set<String> zipRootDirs = new HashSet<>();
         try {
-            if (zipFilename.indexOf(TAR_GZIP_FILE_EXT) > 0) {
+            if (TAR_GZIP_FILE_EXT.equalsIgnoreCase(FilenameUtils.getExtension(zipFilename))) {
```

Comment:
This logic worries me because it's not a 1 for 1 replacement. The original check was looking for any instance of ".gz" which in itself seems error prone, but the new logic checks for a file extension of "gz".
Did you regression test this change already on a Linux deployment?

Or even better, is there a way to check the file format with the Tika library?

Reply:
It's not a 1 to 1 replacement because FilenameUtils.getExtension returns the extension without the dot. e.g. test.tar.gz will return "gz" but test.targz will return blank.

Reply:
Good suggestion on TIKA, it does support detection of gz file. I'll change the code to use gz and write the tests for it.